src/models/user-image-info.ts

```diff
@@ -0,0 +1,124 @@
+import { NumberParser } from '@internetarchive/field-parsers';
+import { UserImageInfoResponse } from './response';
+
+export interface UserImageInfoInterface {
+  /**
+   * The image filename
+   */
+  name?: string;
+
+  /**
+   * The image source, ie `original` or `derivative`
+   */
+  source?: string;
+
+  /**
+   * The last modified epoch
+   */
+  mtime?: number;
+
+  /**
+   * The file size
+   */
+  size?: number;
+
+  /**
+   * The md5 hash
+   */
+  md5?: string;
+
+  /**
+   * The image crc32
+   */
+  crc32?: string;
+
+  /**
+   * The image sha1
+   */
+  sha1?: string;
+
+  /**
+   * The image format. Will likely be "Item Image"
+   */
+  format?: string;
+
+  /**
+   * The image rotation
+   */
+  rotation?: number;
+}
+
+export class UserImageInfo implements UserImageInfoInterface {
+  /** @inheritdoc */
+  name?: string;
+
+  /** @inheritdoc */
+  source?: string;
+
+  /** @inheritdoc */
+  mtime?: number;
+
+  /** @inheritdoc */
+  size?: number;
+
+  /** @inheritdoc */
+  md5?: string;
+
+  /** @inheritdoc */
+  crc32?: string;
+
+  /** @inheritdoc */
+  sha1?: string;
+
+  /** @inheritdoc */
+  format?: string;
+
+  /** @inheritdoc */
+  rotation?: number;
+
+  static fromResponse(userResponse: UserImageInfoResponse): UserImageInfo {
+    const numberParser = NumberParser.shared;
+    let mtime: number | undefined;
+    let size: number | undefined;
+    let rotation: number | undefined;
+
+    if (userResponse.mtime) mtime = numberParser.parseValue(userResponse.mtime);
+    if (userResponse.size) size = numberParser.parseValue(userResponse.size);
+    if (userResponse.rotation)
+      rotation = numberParser.parseValue(userResponse.rotation);
```

Comment:
SUGGESTION: I thought our style preference was for `if` bodies to either be on the same line as the `if`, in which case no braces necessary; or if the body is on another line -- even if 1 line -- then put braces around it. (This is different from our PHP convention.)